lib/broadway/topology.ex

```diff
@@ -151,12 +150,24 @@ defmodule Broadway.Topology do
       processors_config: init_processors_config(opts[:processors]),
       batchers_config: opts[:batchers],
       context: opts[:context],
-      terminator: :"#{name_prefix(opts[:name])}.Terminator",
       max_restarts: opts[:max_restarts],
       max_seconds: opts[:max_seconds],
       shutdown: opts[:shutdown],
       resubscribe_interval: opts[:resubscribe_interval]
     }
+    |> put_terminator()
+    |> put_rate_limiter(opts)
+  end
+
+  defp put_terminator(config) do
+    Map.put(config, :terminator, process_name(config, "Terminator"))
+  end
+
+  defp put_rate_limiter(config, opts) do
+    case get_in(opts, [:producer, :rate_limiting]) do
+      nil -> Map.put(config, :rate_limiter, nil)
+      _ -> Map.put(config, :rate_limiter, process_name(config, "RateLimiter"))
+    end
```

Comment:
If you initialize it to nil above, then:

```suggestion
    if opts[:producer][:rate_limiting] do
      Map.put(config, :rate_limiter, process_name(config, "RateLimiter"))
    else
      config
    end
```